packages/chrome/src/ChromeProvider/ChromeProvider.tsx

```diff
@@ -7,28 +9,70 @@ import { IDENTITY_URL, LAST_VISITED_URL, get, post } from '../utils/fetch';
 
 const getUserIdentity = () => get<UserIdentity>(IDENTITY_URL);
 
-const useLastPageVisitedUploader = (providerState: ReturnType<typeof chromeState>, bundle = '') => {
+const useLastPageVisitedUploader = (providerState: ReturnType<typeof chromeState>) => {
+  const scalprum = useScalprum<{ initialized: boolean; api: { chrome: ChromeAPI } }>();
   const { pathname } = useLocation();
+  const postData = async (pathname: string, title: string, bundle: string) => {
+    try {
+      const data = await post<LastVisitedPage[], { pathname: string; title: string; bundle: string }>(LAST_VISITED_URL, {
+        pathname,
+        title,
+        bundle,
+      });
+      providerState.setLastVisited(data);
+    } catch (error) {
+      console.error('Unable to update last visited pages!', error);
+    }
+  };
   useEffect(() => {
-    post<LastVisitedPage[], { pathname: string; title: string; bundle: string }>(LAST_VISITED_URL, {
-      pathname,
-      title: document.title,
-      bundle,
-    })
-      .then((data) => providerState.setLastVisited(data))
-      .catch((error) => console.error('Unable to update last visited pages!', error));
+    let titleObserver: MutationObserver | undefined;
+    let prevTitle: string | null;
+    const titleTarget = document.querySelector('title');
+    if (titleTarget) {
+      prevTitle = titleTarget.textContent;
+      // initial api call on mount
+      postData(pathname, prevTitle ?? '', scalprum.api.chrome.getBundleData().bundleTitle);
+      /**
+       * Use Mutation observer to trigger the updates.
+       * Using the observer will ensure the last visited pages gets updated on document title change rather than just location change.
+       * The chrome service uses pathname as identifier and updates title according.
+       * Multiple calls with the same pathname and different title will ensure that the latest correct title is assigned to a pathname.       *
+       *  */
+      titleObserver = new MutationObserver((mutations) => {
+        // grab text from the title element
+        const currentTitle = mutations[0]?.target.textContent;
+        console.warn({ prevTitle, currentTitle, pathname });
+        // trigger only if the titles are different
+        if (typeof currentTitle === 'string' && currentTitle !== prevTitle) {
+          try {
+            prevTitle = currentTitle;
+            postData(pathname, currentTitle, scalprum.api.chrome.getBundleData().bundleTitle);
+          } catch (error) {
+            // catch sync errors
+            console.error('Unable to update last visited pages!', error);
```

Comment:
NIT (not blocking) - we have the same error catch/logger on line 24. Might be nice to create a single error handler for errors that occur anywhere in this logic.

Reply:
@florkbr It's not really the same. The only way of being able to use the same logic would be making the observer callback async and I don't want to do that. It can cause some issues. The catch block you have highlighted is there solely for the purpose of catching this call 
```
scalprum.api.chrome.getBundleData().bundleTitle
```
Where as the top one is for the async call. Sure we could create some handle error method but I don't really see a point n wrapping a console.error statement in a function.